bin/propolis-server/src/main.rs

```diff
@@ -35,16 +80,30 @@ enum Args {
         #[clap(name = "PROPOLIS_IP:PORT", action)]
         propolis_addr: SocketAddr,
 
-        /// IP:Port with which to register as an Oximeter metric producer
-        #[clap(long, action)]
-        metric_addr: Option<SocketAddr>,
+        /// Method for registering as an Oximeter metric producer.
+        ///
+        /// The following values are supported:
+        ///
+        /// disable - Do not register or attempt to produce metrics.
+        ///
+        /// dns - Register at an address inferred from Oxide internal DNS.
+        /// This is only available if the Propolis is listening on a
+        /// non-localhost IPv6 address.
+        ///
+        /// IP:port - Register with the explicitly-provided socket address.
+        #[clap(long, default_value_t = MetricRegistration::Disable)]
+        metrics_addr: MetricRegistration,
 
         #[clap(
             name = "VNC_IP:PORT",
             default_value_t = SocketAddr::new(IpAddr::V6(Ipv6Addr::UNSPECIFIED), 5900),
             action
         )]
         vnc_addr: SocketAddr,
+
+        /// Logging level for the server
+        #[clap(long, default_value_t = slog::Level::Info, value_parser = parse_log_level)]
+        log_level: slog::Level,
```

Comment:
Adding this feels like a good idea, but it also seems pretty much completely unrelated to this change --- what's the motivation for doing it as part of this PR?

Reply:
I wanted to see the `oximeter_producer::Server` log messages to confirm that it was appropriately using internal DNS. That's all logged at `debug` or `trace`, so I needed a way to increase verbosity at the top level.